src/query/api/v1/handler/headers.go

```diff
@@ -56,6 +56,9 @@ const (
 	// metrics type.
 	MetricsStoragePolicyHeader = "M3-Storage-Policy"
 
+	// QueryOptionsJSONHeader provides tag options to enforces on queries.
+	QueryOptionsJSONHeader = "M3-Restrict-By-Tags-JSON"
```

Comment:
We should probably update the header name to match how the header reads (to be consistent with other headers defined here).

So `QueryOptionsJSONHeader` -> `RestrictByTagsJSONHeader`?